### PR TITLE
Springer: Added Harvesting configs.

### PR DESCRIPTION
**Pull request by Neodia**

### CODE REVIEW

File: dags/springer/dag_pull_ftp.py [hunk 1]
```diff
@@ -5,22 +5,27 @@
...
         repo: IRepository = SpringerRepository(),
+        **kwargs
     ):
-        pull_ftp.migrate_from_ftp(sftp, repo)
+        return pull_ftp.migrate_from_ftp(sftp, repo, logger, **kwargs)
 
     @task()
-    def trigger_file_processing(repo: IRepository = SpringerRepository()):
-        return pull_ftp.trigger_file_processing("springer", repo)
+    def trigger_file_processing(filenames=[], repo: IRepository = SpringerRepository()):
```
Comment (drjova) [inline dags/springer/dag_pull_ftp.py]: it's not a good practise to have mutable args in python https://florimond.dev/en/posts/2018/08/python-mutable-defaults-are-the-source-of-all-evil/

File: dags/springer/repository.py [hunk 2]
```diff
@@ -44,8 +54,15 @@ def __find_all_extracted_files(self):
...
             for f in self.s3.objects.filter(Prefix=self.EXTRACTED_DIR).all()
-            if self.__file_is_meta(f.key) or ".pdf" in f.key
+            if self.is_meta(f.key) or ".pdf" in f.key
+        ]
+
+    def __find_given_filenames(self, filenames):
+        return [
+            f.key
+            for f in self.s3.objects.filter(Prefix=self.EXTRACTED_DIR).all()
+            if f.key.split("/")[-1] in filenames
```
Comment (drjova) [inline dags/springer/repository.py]: why not try to get the object directly instead of listing everything?

File: dags/common/pull_ftp.py [hunk 3]
```diff
@@ -6,28 +6,85 @@
 from airflow.api.common import trigger_dag
 from common.repository import IRepository
 from common.sftp_service import SFTPService
+from structlog import PrintLogger
 
 
-def migrate_from_ftp(sftp: SFTPService, repo: IRepository):
-    filenames = sftp.list_files()
+def migrate_files(filenames, sftp: SFTPService, repo: IRepository, logger: PrintLogger):
```
Comment (ErnestaP) [inline dags/common/pull_ftp.py:13]: For me this function, looks pretty long, what do you think about splitting it into smaller functions (move them to utils) and using these functions here? Would be easy readable

File: dags/common/pull_ftp.py [hunk 4]
```diff
@@ -6,28 +6,85 @@
...
+    filenames = sftp.list_files()
+    return migrate_files(filenames, sftp, repo, logger)
+
+
+def migrate_from_ftp(
+    sftp: SFTPService, repo: IRepository, logger: PrintLogger, **kwargs
+):
+    filenames = []
+    params = kwargs["params"]
+    if "force_pull" in params and params["force_pull"]:
```
Comment (drjova) [inline dags/common/pull_ftp.py:38]: I would split it in three different functions. Are the `params` passed from airflow? can we just have them as kwargs with defaults?

File: dags/springer/dag_pull_ftp.py (hunk 1 above)
Comment (Neodia) [reply dags/springer/dag_pull_ftp.py]: Ohh yes I remember thtat 🥲

**Review by drjova — APPROVED** [review]

**Review by ErnestaP — APPROVED** [review]

File: dags/springer/repository.py (hunk 2 above)
Comment (Neodia) [reply dags/springer/repository.py]: Done

File: dags/common/pull_ftp.py (hunk 3 above)
Comment (Neodia) [reply dags/common/pull_ftp.py:13]: Done

File: dags/common/pull_ftp.py (hunk 4 above)
Comment (Neodia) [reply dags/common/pull_ftp.py:38]: Done. The params are passed by the user from the Airflow webview. I added defaults in the DAG 👍

**Pull request merged**